src/POData/ObjectModel/ObjectModelSerializer.php

```diff
@@ -653,12 +653,12 @@ private function _writeMediaResourceMetadata(
         if ($resourceType->isMediaLinkEntry()) {
             $odataEntry->isMediaLinkEntry = true;
             $streamProvider = $this->getService()->getStreamProvider();
-            $eTag = $streamProvider->getStreamETag($entryObject, null);
-            $readStreamUri = $streamProvider->getReadStreamUri($entryObject, null, $relativeUri);
-            $mediaContentType = $streamProvider->getStreamContentType($entryObject, null);
+            $eTag = $streamProvider->getStreamETag2($entryObject,null, $this->service->getOperationContext());
```

Comment:
Can you cache the calls to $this->service->getOperationContext() in a local variable?